packages/language-server/src/utils.ts

```diff
@@ -127,3 +127,47 @@ export function debounceThrottle<T extends (...args: any) => void>(fn: T, millis
 
 	return maybeCall as any;
 }
+
+export interface AstroVersion {
+	full: string;
+	major: number;
+	minor: number;
+	patch: number;
+	exist: boolean;
+}
+
+export function getUserAstroVersion(basePath: string): AstroVersion {
+	let version = '0.0.0';
+	let exist = true;
+
+	try {
+		const astroPackageJson = require.resolve('astro/package.json', { paths: [basePath] });
+
+		version = require(astroPackageJson).version;
+	} catch {
+		// If we couldn't find it inside the workspace's node_modules, it might means we're in the monorepo
+		try {
+			const monorepoPackageJson = require.resolve('./packages/astro/package.json', { paths: [basePath] });
+			version = require(monorepoPackageJson).version;
+		} catch (e) {
+			// If we still couldn't find it, it probably just doesn't exist
+			exist = false;
+			console.error(e);
+		}
+	}
+
+	let [major, minor, patch] = version.split('.');
```

Comment:
We very much could! I just felt like since we didn't do anything too complicated (we don't really need to validate the version or anything), it wasn't necessary but if you think that it's better, I could use `node-semver` or similar

For context, this is also how Svelte's language-tools parse versions of packages so this solution is "battle-tested"

> I'm not totally sure what is trying to be achieved with this code

Do you think it needs additional comments? I can document it better for sure

Reply:
The above response about the beta makes me understand the use-case here, so I don't think adding this is necessary.